### PR TITLE
Add notes about versionadded/versionchanged directives to dev guide

### LINKED ISSUE #7192: Add .. versionadded:: directives when modifying APIs
## 📚 Documentation

The versionadded directive is a really nice way to tell users when particular arguments or functions were added to an API (example [here](https://github.com/scikit-image/scikit-image/blob/8ba3bfffa79f24810e3be8e5f833ea7e0f954750/skimage/filters/_median.py#L31)). We haven't been using them in napari, which leads to some confusion about when particular APIs were added (who has time to read through a bunch of release notes?) and therefore what the minimum version of napari is to be able to use those APIs (ex: https://github.com/BiAPoL/napari-clusters-plotter/pull/308#issuecomment-2285460086).

### PR DESCRIPTION
**Pull request by melissawm**
# References and relevant issues
Closes napari/napari#7192
Complements napari/napari#7359

# Description
Adds note about using versionadded/versionchanged directives to the core dev and contributing guides.

### CODE REVIEW

File: docs/developers/contributing/index.md
Comment (lucyleeow) [inline docs/developers/contributing/index.md]:
```suggestion
If you are changing an existing API or adding a new one, make sure you update
```

File: docs/developers/coredev/core_dev_guide.md
Comment (lucyleeow) [inline docs/developers/coredev/core_dev_guide.md:104]: Glad to have this documented ! The other question, should napari always be in backticks?

**Review by lucyleeow — APPROVED** [review]
Thank you!

Do you think it's worth adding an example? E.g.,

```python
class Layer(KeymapProvider, MousemapProvider, ABC, metaclass=PostInit):
    """Base layer class.

    Parameters
    ----------
    data : array or list of array
        Data that the layer is visualizing. Can be N-dimensional.

        ... versionchanged::
            Data can now be a list of arrays.
```

this was probably a bad example as this wasn't changed, I don't know what API has been changed recently.

File: docs/developers/coredev/core_dev_guide.md
Comment (lucyleeow) [reply docs/developers/coredev/core_dev_guide.md:104]: I wonder if this part should be added somewhere more generally for all users (not just core devs) e.g., in our guide to contributing documentation?

Comment (melissawm) [reply docs/developers/coredev/core_dev_guide.md:104]: No I think this was meant to show how you should write napari in the sources.

I thought about it and decided against adding this to the general "contributing docs" pages for two reasons. One is that if you are contributing to the docs in general you are not going to be changing APIs so you will not face this case. The other is that I see adding this directive in general as the job of the core team, in that you should probably suggest this on review same as you would do for adding a release note, for example. So as long as the core team is informed, I think we are good.

Comment (lucyleeow) [reply docs/developers/coredev/core_dev_guide.md:104]: Thats a good point. I can see both sides. Pros of adding it to our general contributing guide (agree, docs is not the right place, should be in code contributing docs e.g., if you are changing the API ...):

* it highlights the importance of documentation (this is the culture we want!)
* I think I'd prefer not to add 'extra work' to maintainers and not put the onus on reviewer to check this. We could suggest to contributors that this should be added, but maintainers can handle it if they prefer

BUT I get that we may want to make it easier for contributors, and default let maintainers handle this.

Happy either way.

Comment (melissawm) [reply docs/developers/coredev/core_dev_guide.md:104]: So do you have a suggestion on the wording to use instead of the note I added above to the contributing/index doc?

File: docs/developers/contributing/index.md
Comment (psobolewskiPhD) [inline docs/developers/contributing/index.md]:
```suggestion
[Sphinx documentation](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#describing-changes-between-versions). 
Please also consider documenting any major features/changes in our
[tutorials](tutorials) and other [usage documentation](usage).
```
How about this as a middle ground? encourage but not require?
At the very least would be nice to have issues created that a new feature requires docs -- this is something we should do in review.

**Review by psobolewskiPhD — APPROVED** [review]
Small suggestion, based on the discussion.

File: docs/developers/contributing/index.md
Comment (melissawm) [reply docs/developers/contributing/index.md]: Oh! I think I misunderstood @lucyleeow 's suggestion. I thought she meant adding a note about the versionadded/versionchanged directives in the "contributing docs" section, not the other way around. Sure, this is reasonable!

Comment (psobolewskiPhD) [reply docs/developers/contributing/index.md]: I'll let @lucyleeow merge (feel free) or mark ready-to-merge, in case I also misunderstood.

Comment (lucyleeow) [reply docs/developers/contributing/index.md]: Yup that's exactly what I was thinking, thanks everyone, I'll merge!

**Pull request merged**
